component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -311,21 +346,34 @@ protected void processAuthenticationResponse(HttpServletRequest request,
 
         diagnosticLog.info("Processing TOTP authentication response.");
         String token = request.getParameter(TOTPAuthenticatorConstants.TOKEN);
-        String username = context.getProperty("username").toString();
+        AuthenticatedUser authenticatingUser =
+                (AuthenticatedUser) context.getProperty(TOTPAuthenticatorConstants.AUTHENTICATED_USER);
+        String username = UserCoreUtil.addTenantDomainToEntry(authenticatingUser.getUserName(),
+                context.getTenantDomain());
+        username = UserCoreUtil.addDomainToName(username, authenticatingUser.getUserStoreDomain());
```

Comment:
can't we use authenticatingUser.toFullQualifiedUsername

Reply:
Fixed https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/105/commits/9ea77f5434ce5d211d19c714fc2f09fadf192a88